Mage.Tests/src/test/java/org/mage/test/cards/rules/DontUntapTest.java

```diff
@@ -73,5 +73,113 @@ public void TestAjaniVengeantFirst() {
         assertTapped("Sublime Archangel", true);
 
     }
+     
+    /**
+     * Issue #10746: Incorrect implementations for Misstep; Pollen Lullaby; Icebreaker Kraken
+     */
+    @Test
+    public void testExhaustionEffect() {
+        addCard(Zone.BATTLEFIELD, playerA, "Silvercoat Lion", 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Swamp", 2);
+        addCard(Zone.GRAVEYARD, playerA, "Reassembling Skeleton", 1);
+
+        addCard(Zone.BATTLEFIELD, playerB, "Island", 3);
+        addCard(Zone.HAND, playerB, "Exhaustion", 1);
+        
+        attack(1, playerA, "Silvercoat Lion");
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, "Exhaustion", playerA);
+        activateAbility(2, PhaseStep.END_TURN, playerA, "{1}{B}:"); //Reassembling Skeleton enters after spell is cast, should remain tapped
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
```

Comment:
Make sure to use `setStrictChooseMode(true)` on all new tests (to guard against unintended behavior that is silently corrected by AI auto choose).